tests/test_py_result_uploader.py

```diff
@@ -0,0 +1,219 @@
+
+    def test_skip(self, single_skip_xml, properties):
+        """Verify that a valid qTest 'AutomationTestLogResource' swagger model is generated from a JUnitXML file
+        that contains a single skipping test
+        """
+
+        # Setup
+        junit_xml = py_result_uploader._load_input_file(single_skip_xml)
+        # noinspection PyUnresolvedReferences
+        test_log_dict = py_result_uploader._generate_test_log(junit_xml.find('testcase'), properties).to_dict()
+
+        # Expectation
+        test_name = 'test_skip'
+        test_log_exp = {'name': test_name,
+                        'status': 'SKIPPED',
+                        'module_names': [properties['GIT_BRANCH']],
+                        'automation_content': '{}#{}'.format(properties['GIT_BRANCH'], test_name)}
+
+        # Test
+        for exp in test_log_exp:
+            assert test_log_exp[exp] == test_log_dict[exp]
+
+
+class TestGenerateAutoRequest(object):
+    """Test cases for the '_generate_auto_request' function"""
+
+    def test_mix_status(self, flat_mix_status_xml):
+        """Verify that a valid qTest 'AutomationRequest' swagger model is generated from a JUnitXML file
+        that contains multiple tests with different status results
+        """
+
+        # Setup
+        test_cycle = 'CL-1'
+        junit_xml = py_result_uploader._load_input_file(flat_mix_status_xml)
+        # noinspection PyUnresolvedReferences
+        auto_req_dict = py_result_uploader._generate_auto_request(junit_xml, test_cycle).to_dict()
+
+        # Expectation
+        prop_value = 'Unknown'
+        test_logs_exp = [{'name': 'test_pass',
+                          'status': 'PASSED',
+                          'module_names': [prop_value],
+                          'automation_content': '{}#{}'.format(prop_value, 'test_pass')},
+                         {'name': 'test_fail',
+                          'status': 'FAILED',
+                          'module_names': [prop_value],
+                          'automation_content': '{}#{}'.format(prop_value, 'test_fail')},
+                         {'name': 'test_error',
+                          'status': 'FAILED',
+                          'module_names': [prop_value],
+                          'automation_content': '{}#{}'.format(prop_value, 'test_error')},
+                         {'name': 'test_skip',
+                          'status': 'SKIPPED',
+                          'module_names': [prop_value],
+                          'automation_content': '{}#{}'.format(prop_value, 'test_skip')}]
+
+        # Test
+        for x in range(len(auto_req_dict['test_logs'])):
+            for key in test_logs_exp[x]:
+                assert test_logs_exp[x][key] == auto_req_dict['test_logs'][x][key]
+
+
+class TestOutputJsonAutoRequest(object):
+    """Test cases for the 'output_json_auto_request' function"""
+
+    def test_happy_path(self, single_passing_xml, tmpdir_factory):
```

Comment:
How does Python handle mirrored method names / param names.  In Ruby Rubocop would complain about this.